app/Model/Color.php

```diff
@@ -108,9 +124,12 @@ class Color extends Base
      */
     public function find($color)
     {
+        if (!$this->colors) {
+            $this->loadColors();
```

Comment:
Instead of calling multiple times the method `loadColors()` you can add constructor that override the parent one and then load the colours from the database.

Example:

```php
public function __construct(Container $container)
{
    parent::__construct($container);
    $this->loadFromDatabase();
}
```